danger/src/ChangelogChecker.ts

```diff
@@ -0,0 +1,324 @@
+  const dangerBaseRef = pr.head.ref;
+  const dangerMessage = `Add missing changelog`;
+  const dangerTags = `[danger][bot]`;
+
+  const fileMap = missingEntries.reduce(
+    (prev, current) => ({
+      ...prev,
+      [getPackageChangelogPath(current.packageName)]: current.content,
+    }),
+    {}
+  );
+
+  await github.createOrUpdateBranchFromFileMap(fileMap, {
+    baseBranchName: dangerBaseRef,
+    branchName: dangerHeadRef,
+    message: `${dangerTags} ${dangerMessage}`,
+  });
+
+  const prs = await github.getOpenPRs({
+    fromBranch: dangerHeadRef,
+    toBranch: dangerBaseRef,
+  });
+
+  if (prs.length > 1) {
+    warn("Couldn't find a correct pull request. Too many open ones.");
+    return null;
+  }
+
+  if (prs.length === 1) {
+    return prs[0].html_url;
+  }
+
+  const { html_url } = await github.openPR({
+    fromBranch: dangerHeadRef,
+    toBranch: dangerBaseRef,
+    title: `${dangerTags} ${dangerMessage} to #${pr.number}`,
+    body: `${dangerMessage} to #${pr.number}`,
+  });
+
+  return html_url;
+}
+
+function generateReport(
+  missingEntries: Array<{ packageName: string; diff: string }>,
+  url?: string | null
+) {
+  const message = missingEntries
+    .map(
+      entry =>
+        `- <code>${danger.github.utils.fileLinks(
+          [getPackageChangelogPath(entry.packageName)],
+          false
+        )}</code>`
+    )
+    .join('\n');
+
+  const diff = '```diff\n' + missingEntries.map(entry => entry.diff).join('\n') + '```\n';
+  const pr = url ? `#### or merge this pull request: ${url}` : '';
+  fail(`
+📋 **Missing Changelog**
+------
+🛠 Add missing entires to:
+${message}`);
+
+  markdown(
+    `
+### 🛠 Suggested fixes:
+
+<details>
+  <summary>📋 Missing changelog</summary>
+
+  #### Apply suggested changes:
+${diff}
+${pr} 
+</details>`
+  );
+}
+
+/**
+ * This function checks if the changelog was modified, doing the following steps:
+ * - get packages which were modified but don't have changes in `CHANGELOG.md`
+ * - parse PR body to get suggested entries for those packages
+ * - run `et add-changelog` for each package to apply the suggestion
+ * - create a new PR
+ * - add a comment to inform about missing changelog
+ * - fail CI job
+ */
+export async function checkChangelog(): Promise<void> {
+  const modifiedPackages = groupBy(
+    modifiedFiles.filter(file => file.startsWith('packages')),
+    file => file.split(path.sep)[1]
+  );
+
+  const packagesWithoutChangelog = Object.keys(modifiedPackages).filter(
+    packageName => !isChangelogModified(packageName, modifiedPackages[packageName])
+  );
+  if (packagesWithoutChangelog.length === 0) {
+    message(`✅ **Changelog**`);
+    return;
+  }
+
+  const suggestedEntries = getSuggestedChangelogEntries(packagesWithoutChangelog);
+  const fixedEntries = await runAddChangelogCommand(suggestedEntries);
+
+  const url = await createOrUpdateRP(fixedEntries);
+  await generateReport(fixedEntries, url);
```

Comment:
What would you say for splitting this file named `ChangelogChecker` into multiple ones?
I suggest extracting `PullRequestManager`, `ChangelogChecker` and some of functions might land in `Utils` (these `git`-related and `file`-related)?